R/ppm.R

```diff
@@ -33,35 +33,34 @@ renv_ppm_transform_impl <- function(url) {
   if (grepl("/__[^_]+__/", url))
     return(url)
 
+  # try to parse the repository URL
+  parts <- catch(renv_url_parse(url))
+  if (inherits(parts, "error"))
+    return(url)
+
   # only attempt to transform URLs that are formatted like PPM urls:
   #
   #   https://ppm.company.org/cran/checkpoint/id
   #
   # in particular, there should be at least two trailing
   # alphanumeric path components
-  pattern <- "/[^/]+/[^/]+/"
-  if (!grepl(pattern, url))
+  pattern <- "/[^/]+/[^/]+"
+  if (!grepl(pattern, parts$path))
     return(url)
 
-  # begin building list of URLs
-  urls <- character()
-
-  # ignore some known CRAN mirrors
-  # (note that getCRANmirrors can fail if R is not installed with
-  # any local knowledge of available CRAN repositories)
+  # check if this is an 'ignored' URL; that is, a repository which we
+  # know is not a PPM URL
   mirrors <- catch(getCRANmirrors(local.only = TRUE))
-  urls <- c(urls, mirrors$URL)
-
-  # also ignore some RStudio URLs
-  rstudio <- c(
+  ignored <- c(
+    getOption("renv.ppm.ignoredUrls", default = character()),
```

Comment:
Need to document `renv.ppm.ignoredUrls` somewhere?

Reply:
Or should I just add it to config.yml? Or should it be a project-specific setting?

Reply:
Seems like project specific might be most useful?

Reply:
I'll do this in a separate PR (I think the other PPM stuff I added should also be project specific)